mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -419,7 +421,8 @@ def create_model_version(
         notebook_id = self._get_notebook_id(run)
         extra_headers = None
         if notebook_id is not None:
-            extra_headers = {_DATABRICKS_LINEAGE_ID_HEADER: str(notebook_id)}
+            entity_info = EntityInfo(entity_type=NOTEBOOK, entity_id=str(notebook_id))
+            extra_headers = {_DATABRICKS_LINEAGE_ID_HEADER: entity_info.SerializeToString()}
```

Comment:
Have we tested / could we test that this is actually JSON? I don't see us calling this API elsewhere in the MLflow codebase - let me see what we do elsewhere

Reply:
Typically we call `message_to_json` to convert to JSON [e.g. here](https://github.com/mlflow/mlflow/blob/67ed9b70eaf7a99c6bc60bd276a91273d5454698/mlflow/store/_unity_catalog/registry/rest_store.py#L403-L411), it would probably be best to be consistent here

Reply:
So this does not return JSON but the protobuf encoded string meant for transferring the message over the wire:

https://protobuf.dev/programming-guides/encoding/

Is there a reason you want raw json to be written here?

Reply:
I think it might also need to be this: 

`base64.b64encode(entity_info.SerializeToString()).decode("utf-8")`

Reply:
> Is there a reason you want raw json to be written here?

Yeah, we discussed JSON-serializing this header to make it easier for callers to pass

Reply:
BTW It's also possible that we'll want to allow callers to pass info besides an `EntityInfo` in the header value in the future - I was thinking if we should instead wrap it in something like a `message LineageHeaderInfo` with field `entity_info` that could later become a `oneof` to allow passing other things. 

```
message LineageHeaderInfo {
  oneof info {
    EntityInfo entity_info = 1;
    // As one possibility, it might be possible in the future for the edge between two lineage nodes to also be a securable
   // e.g. if notebooks or jobs are moved into UC
    SecurableReference securable_reference = 2;
  }
}  
```

But at least if we JSON-serialize the header value, I think we have the flexibility to do something like [what's done here](https://databricks.slack.com/archives/C01AC07DWD6/p1657673980940779) to support passing something other than the EntityInfo that we directly inject here.

Reply:
>Yeah, we discussed (with Menglei and Gopal) JSON-serializing this header value to make it easier for callers to pass

I guess I misunderstood that part then.  What is the benefit of doing this as JSON over leveraging protobuf encoding?  I just checked in universe code and we definitely deser from encoded strings all over the place.  Unless we have a solid precedent of using JSON objects in these HTTP headers, I'm not seeing a reason to use JSON?

Reply:
> BTW It's also possible that we'll want to allow callers to pass info besides an `EntityInfo` in the header value in the future - I was thinking if we should instead wrap it in something like a `message LineageHeaderInfo` with field `entity_info` that could later become a `oneof` to allow passing other things.

This goes back to what we wanted the format of this header to be and we had seemed to agree that the EntityInfo proto definition made sense.  I'm not against a wrapper JSON object but we should clarify that with Menglei because discovery team is going to have to be the people that own this definition.


> But at least if we JSON-serialize the header value, I think we have the flexibility to do something like [what's done here](https://databricks.slack.com/archives/C01AC07DWD6/p1657673980940779) to support passing something other than the EntityInfo that we directly inject here.

Again, I'm not against this, but this is a separate discussion from whether we serialize as raw JSON or a protobuf encoded string.